### PR TITLE
feat: cancel subscription

### PR DESCRIPTION
**Pull request by Ipriyankrajai**
fixes: #153

### CODE REVIEW

File: apps/web/utils/actions.ts [hunk 1]
```diff
@@ -106,6 +125,9 @@ export async function deleteAccountAction() {
     captureException(error);
   }
 
+  if (lemonSqueezySubscriptionItemId)
+    await cancelSubScriptionForUser(lemonSqueezySubscriptionItemId);
+
   await prisma.user.delete({ where: { email: session.user.email } });
```
Comment (Ipriyankrajai) [inline apps/web/utils/actions.ts]: I think we should delete the user after the subscription tenure is ended.

File: apps/web/utils/actions.ts [hunk 2]
```diff
@@ -89,6 +92,22 @@ export async function deleteAccountAction() {
...
   if (!session?.user.email) throw new Error("Not logged in");
 
+  const user = await prisma.user.findUniqueOrThrow({
+    where: { id: session.user.id },
+    select: {
+      premium: {
+        select: {
+          id: true,
+          tier: true,
+          lemonSqueezySubscriptionItemId: true,
```
Comment (elie222) [inline apps/web/utils/actions.ts]: I don't think we need any of these.

We need: `lemonSqueezySubscriptionId` (which is different to `lemonSqueezySubscriptionItemId`)

File: apps/web/utils/actions.ts (hunk 1 above)
Comment (elie222) [reply apps/web/utils/actions.ts]: No, don't delete the user. "leaving a user's data in Stripe after they delete their account is generally allowed under GDPR, provided that you have a legitimate reason for retaining this data, such as for financial records, compliance with tax law, or other regulatory requirements."

**Review by elie222 — CHANGES_REQUESTED** [review]
Merge conflict too :(

File: apps/web/utils/actions.ts (hunk 2 above)
Comment (Ipriyankrajai) [reply apps/web/utils/actions.ts]: Done

File: apps/web/utils/actions.ts (hunk 1 above)
Comment (Ipriyankrajai) [reply apps/web/utils/actions.ts]: For now i have commented that part

File: apps/web/utils/actions.ts [hunk 3]
```diff
@@ -92,6 +93,21 @@ export async function deleteAccountAction() {
   const session = await auth();
   if (!session?.user.email) throw new Error("Not logged in");
 
+  const user = await prisma.user.findUniqueOrThrow({
+    where: { id: session.user.id },
+    select: {
+      premium: {
+        select: {
+          id: true,
+          tier: true,
```
Comment (elie222) [inline apps/web/utils/actions.ts:102]: We don't need `id` or `tier` here, right?

File: apps/web/utils/actions.ts [hunk 4]
```diff
@@ -109,7 +125,10 @@ export async function deleteAccountAction() {
     captureException(error);
   }
 
-  await prisma.user.delete({ where: { email: session.user.email } });
+  if (lemonSqueezySubscriptionId)
+    await cancelSubScriptionForUser(lemonSqueezySubscriptionId);
+
+  // await prisma.user.delete({ where: { email: session.user.email } });
```
Comment (elie222) [inline apps/web/utils/actions.ts:131]: Uncomment. We do want to delete the user.

(We do want to keep the user in Lemon though)

**Review by elie222 — CHANGES_REQUESTED** [review]
Left 2 comments

**Pull request closed without merging**